backend/src/main/java/hu/flowacademy/munkalap/controller/UserController.java

```diff
@@ -0,0 +1,42 @@
+package hu.flowacademy.munkalap.controller;
+
+import hu.flowacademy.munkalap.dto.UserCreateDTO;
+import hu.flowacademy.munkalap.entity.User;
+import hu.flowacademy.munkalap.exception.WorksheetUserException;
+import hu.flowacademy.munkalap.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.annotation.security.RolesAllowed;
+
+@RestController
+@RequestMapping("api")
+public class UserController {
+
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/users")
+    @ResponseStatus(HttpStatus.CREATED)
+    @RolesAllowed("admin")
+    public User createUser(@RequestBody UserCreateDTO userCreateDTO) {
+        User user = User.builder()
+                .name(userCreateDTO.getName())
+                .password(userCreateDTO.getPassword())
+                .email(userCreateDTO.getEmail())
+                .build();
+        try {
```

Comment:
Ne legyen try-catch blokk

Reply:
Ha nem tetszik , akkor pls javasolj valamit, nekem pl nincs gondom vele, a service réteg exception-jei mappelődnek itt HttpStatus kódra. Ha mást szeretnénk, akkor kerüljön ide javaslat. Nekem +1 (https://www.baeldung.com/exception-handling-for-rest-with-spring)

Reply:
Jogos, egy későbbi commentben fejtettem ki. Viszont akkor dobjuk tovább itt is és legyen egy exception handler osztály. Tökéletes példa, amit linkeltél.